### DISCUSSION TITLE
How to use jsforce to make composite api requests?

### DISCUSSION THREAD

**Discussion by olliebatch** [jsforce/jsforce#1594] [Q&A]
Hi, 

I'm looking to do an update/create request and followed by a get request. Ideally this would be done in a composite api request as then it won't use multiple api calls in order to fetch this data. 

I can't see any docs on the site or mention of the composite api in general?
Would the only way to be to manually craft requestPosts? 

`  const ret = connection.sobject("Account").findOne({ Id: "1" }).update({
    Id,
    FirstName: "Test1234567",
  });`
  
  I have generally been using the conn.sobject approaches, but can't find a way to chain on those for this use case

**Comment by gmillerd** [comment]
How is a fetch also doing an update? You know this just is a builder right?

Why is there a NEED to have this on a single chain? You could wrangle this into a map async/await function.

Where is your exception handling occuring for this long stack when you chain all this together?

**Reply by olliebatch** [reply]
This was a poorly crafted original request. 
The find one and update was just following the documentation.
Yes i know it is a builder.

The need to have on a single chain is that a composite api doesn't use multiple api calls to handle the same data set. E.G for large processing api's this can be a substantial saving.

For example this below might show that you can do it via the composite api.
I was merely enquiring whether the builder has a way to call these composite api's provided by salesforce. Rather than singular query's

```

 // Use Composite API to update and refetch the record
  conn.requestPost('/services/data/v56.0/composite', {
    "compositeRequest": [
      {
        "method": "PATCH",
        "url": `/services/data/v56.0/sobjects/Account/${recordId}`, // The record you want to update
        "referenceId": "UpdateAccount",
        "body": updateData
      },
      {
        "method": "GET",
        "url": `/services/data/v56.0/sobjects/Account/${recordId}`, // Refetching the updated record
        "referenceId": "GetUpdatedAccount",
        "after": "UpdateAccount" // Ensure the GET request happens after the PATCH
      }
    ]
  }, function(err, result) {
    if (err) {
      return console.error(err);
    }

    // The refetched record data will be in result.compositeResponse
    console.log('Updated and Refetched Record:', result.compositeResponse[1].body);
  });
});


```

**Comment by cristiand391** [comment]
Hey, sorry for the late reply.

> Would the only way to be to manually craft requestPosts?

Yes, there's no option in `conn.sobject` methods to allow chained methods to be combined into one Composite API request.
